lib/mercury/mercury.rb

```diff
@@ -60,17 +60,36 @@ def publish(source_name, msg, tag: '', headers: {}, &k)
     # The amqp gem caches exchange objects, so it's fine to
     # redeclare the exchange every time we publish.
     with_source(source_name) do |exchange|
-      payload = write(msg)
-      pub_opts = Mercury.publish_opts(tag, headers)
-      if publisher_confirms_enabled
-        expect_publisher_confirm(k)
-        exchange.publish(payload, **pub_opts)
-      else
-        exchange.publish(payload, **pub_opts, &k)
-      end
+      publish_internal(exchange, msg, tag, headers, &k)
+    end
+  end
+
+  ORIGINAL_TAG_HEADER = 'Original-Tag'.freeze
```

Comment:
Because they're only used in one place, I would prefer to inline them. Unfortunately, ruby would then allocate a new string each time. Freezing it in place wouldn't help, since ruby would first allocate a new string in order to freeze it! Plus, it would incur the additional cost of interning the string each time.